libsqlite3-sys/build.rs

```diff
@@ -81,6 +97,93 @@ mod build_bundled {
             .flag("-DHAVE_USLEEP=1")
             .warnings(false);
 
+        if cfg!(feature = "bundled-sqlcipher") {
+            cfg.flag("-DSQLITE_HAS_CODEC").flag("-DSQLITE_TEMP_STORE=2");
+
+            let target = env::var("TARGET").unwrap();
+            let host = env::var("HOST").unwrap();
+
+            let is_windows = host.contains("windows") && target.contains("windows");
+            let is_apple = host.contains("apple") && target.contains("apple");
+
+            let lib_dir = env("OPENSSL_LIB_DIR").map(PathBuf::from);
+            let inc_dir = env("OPENSSL_INCLUDE_DIR").map(PathBuf::from);
+            let mut use_openssl = false;
+
+            let (lib_dir, inc_dir) = match (lib_dir, inc_dir) {
+                (Some(lib_dir), Some(inc_dir)) => {
+                    use_openssl = true;
+                    (lib_dir, inc_dir)
+                }
+                (lib_dir, inc_dir) => {
+                    match find_openssl_dir(&host, &target) {
+                        None => {
+                            if is_windows {
+                                panic!("Missing environment variable OPENSSL_DIR or OPENSSL_DIR is not set")
+                            } else if is_apple && Path::new("/opt/local/lib/libssl.a").exists() {
```

Comment:
Macports is not as widely used as e.g. brew.

But... I also am super skeptical about hardcoding this path, as various package managers have been known to leave stale copies of these around (there was a recent homebrew bug about this!)

In general, I'd *really strongly* prefer if we could leave searching for openssl and such to the e.g. openssl-sys crate and the like?

Reply:
I'm happy to drop the convenience lookups for MacPorts, and Homebrew too. The latter came from [rusqlcipher](https://lib.rs/crates/rusqlcipher), which was the basis for my PR, but stalled around 2018. We should give its author(s) credit though for the starting point of this PR.

Once we stabilize on the implementation of this, we can expand the docs to give suggestions about what env variables need to be set for linking against which copies of sqlite3 and/or sqlcipher and/or various crypto provider libraries for sqlcipher.

Taking out the hard-coded convenience paths for openssl is easy. Shifting the search for openssl libraries, when non-bundled, isn't, because there's no package to shift the burden over to. (`openssl-sys` is only a dependency when we're asking for the bundled ssl.) Do you want to drop the `pkg-config` calls too? Would you prefer to shift all of the bundling logic to a new package, which libsqlite3-sys has as an optional dependency, and just retrieve the linking information from `DEP_XXX` flags that the new package sets? I expect that'd be do-able, but more cumbersome and complicated than what we're looking at now.

Reply:
> We should give its author(s) credit though for the starting point of this PR.

A "co-authored-by" line in the final commit maybe would not be out of place, then. Unless you have a better idea.

Probably after you're all done squash this down a bit to clean up the history and add that.

Reply:
Sounds good. I'll wait until it's clear we've got the PR where you want it.